activestorage/lib/active_storage/service.rb

```diff
@@ -141,7 +143,19 @@ def public?
       @public
     end
 
+    def check_if_environment_is_permitted!
+      raise_incorrect_environment! unless permitted_in_current_environment?
+    end
+
     private
+      def raise_incorrect_environment!
+        raise(ActiveStorage::EnvironmentMismatchError.new(service_name, environment))
+      end
+
+      def permitted_in_current_environment?
+        environment.blank? || environment.split(",").include?(Rails.env)
```

Comment:
Should `strip` be called after `split` in the event this comma separated list has white space?

ie

```
    amazon:
      service: S3
      environment: production, staging, demo


environment.blank? || environment.split(",").map(&:strip).include?(Rails.env)
```

Also, would it be better to have the environment key be either a string for a single environment or an array in the yaml? That would eliminate any possible awkwardness of having to remember that `environment` takes a delimitated list.

Exmple:

```
    local:
      service: Disk
      environment: development
      # ...
    amazon:
      service: S3
      environment: 
        - production
        - staging
```

Reply:
Oh yeah that's a much better idea.

Reply:
Thanks @t27duck !